assets/blocks/course-outline/course-block/edit.js

```diff
@@ -84,7 +81,43 @@ const EditCourseOutlineBlock = ( {
 		}
 	}, [ structure, setBlocks, attributes.isPreview ] );
 
-	useSynchronizeLessonsOnUpdate( clientId );
+	useSynchronizeLessonsOnUpdate( clientId, attributes.isPreview );
+
+	// A function which returns all the modules of the course.
+	const getModules = useSelect(
```

Comment:
Does this function have the same purpose as [this one](https://github.com/Automattic/sensei/pull/3677/files#diff-95306f421e247e4a2aee9c369ea0faae74f3ac98d630c55a0f29b76b31ed4f2bR22)? Could we deduplicate these?

cc @yscik

Reply:
Yep that's the method that I based mine on. I wasn't sure of the how we would go with the other PR that's why I didn't look into writing a common one. Done in e67ee0d48f349a919b37c5e1f17d092e73ceed00